packages/mdc-menu/foundation.ts

```diff
@@ -134,12 +133,15 @@ export class MDCMenuFoundation extends MDCFoundation<MDCMenuAdapter> {
   /**
    * Selects the list item at `index` witnin the specified selection group.
    * @param index Index of list item within the selection group.
-   * @param selectionGroupIndex Index of selection group within the menu.
    */
-  setSelectedIndex(index: number, selectionGroupIndex = 0) {
-    const selectionGroup = this.adapter_.getSelectionGroupAtIndex(selectionGroupIndex);
+  setSelectedIndex(index: number) {
+    const listItem = this.adapter_.getListItemByIndex(index);
+    if (!listItem) {
```

Comment:
We should do validation on the `index` number instead similar to List's setSelectedIndex foundation API.

Reply:
are you referring to 
```
  private isIndexValid_(index: MDCListIndex) {
    if (index instanceof Array) {
      if (!this.isCheckboxList_) {
        throw new Error('MDCListFoundation: Array of index is only supported for checkbox based list');
      }

      if (index.length === 0) {
        return true;
      } else {
        return index.some((i) => this.isIndexInRange_(i));
      }
    } else if (typeof index === 'number') {
      if (this.isCheckboxList_) {
        throw new Error('MDCListFoundation: Expected array of index for checkbox based list but got number: ' + index);
      }
      return this.isIndexInRange_(index);
    } else {
      return false;
    }
  }
```

If so I can make this a public static method, or transform it to a util.

Reply:
Yes, but we just need to validate if index is in range (similar to what `isIndexInRange_` method does). Since we can't share the adapters between two components I would just add separate implementation in Menu as of now with custom error message catered to `MDCMenuFoundation`.

Reply:
are you saying we should support multi select for the menu? Otherwise the isIndexValid method should probably just look like:
```
  private isIndexInRange_(index: number) {
    const listSize = this.adapter_.getListItemCount();
    return index >= 0 && index < listSize;
  }

```

We'll also need to add this listItemCount adapter method.

Reply:
Multi-selection is out of scope for Menu. 

Yes, that's the validation I'm referring to, you can make use of existing adapter `getMenuItemCount` to get the count instead.